src/loops/simple_programs_with_repetition/hard/q4/Reduceros.java

```diff
@@ -0,0 +1,64 @@
+package loops.simple_programs_with_repetition.hard.q4;
+
+/*
+Write a program to display a "single-digit reduction" of a number taken from the user by adding up its digits until
```

Comment:
We need to incorporate what to do with negative numbers in this question. If the question works with negative numbers, we should specify to include them and test them. If not, or if you think it doesn't make sense, we should add a test for them and they can be invalid input.